index.js

```diff
@@ -1,9 +1,54 @@
-import {NativeModules} from 'react-native';
+import {NativeModules, Platform, PermissionsAndroid} from 'react-native';
+
+const isAndroid = Platform.OS === 'android';
+
+const defaultTitle = "Request Camera Permission";
+const defaultMessage = "Grant access to enable calling your phone";
 
 var RNImmediatePhoneCall = {
-  immediatePhoneCall: function(number) {
-        NativeModules.RNImmediatePhoneCall.immediatePhoneCall(number);
+  immediatePhoneCall: async function(number, checkPermissions = true, title = defaultTitle, message = defaultMessage, buttonNegative = 'Cancel', buttonPositive = 'OK') {
```

Comment:
I though permissions were supposed to be handled by the app and not this module. Right now as it is, we will not be able to handle internationalization since strings are in the module and not in our app.

I think this whole permission stuff should be handled in the app and only if we have permissions to call, then we call immediatePhoneCall.